org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/AbstractGenerator.java

```diff
@@ -171,13 +171,12 @@ public boolean containsPoint(IPosition point) {
 
 	@Override
 	public Collection<Object> getRegions() {
-		if (regions!=null) return regions;
-		return null;
+		return regions;
 	}
 
 	@Override
 	public void setRegions(Collection<Object> regions) throws GeneratorException {
-		this.regions = regions;
+		this.regions = regions == null ? new ArrayList<Object>() : regions;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)